### ISSUE TITLE
Catching tkinter exception

### ISSUE THREAD

**Issue by ppmm9999** [Delgan/loguru#360]
Hi.
I'm trying to use your library in my little application for all errors and exception. My test code is (look at "a=234/0" - it's raising ZeroDivisionError for test):
```python
import sys
from tkinter import *
from loguru import logger

logger.remove()
logger.add('error.log', mode='w', encoding='UTF8', level='ERROR')

if __debug__:
	logger.add(sys.stderr, level='TRACE')
logger.enable("")

class RLCExcelFile():
	def __init__(self):
		a=234/0
class Converter(Tk):
	def __init__(self):
		Tk.__init__(self)
		self.formats=''
		self.button_start=Button(self, text="Start", command=self.conversion_start)
		self.button_start.pack()
		#self.conversion_start()
	def conversion_start(self):
		self.formats=RLCExcelFile()

@logger.catch
def main():
	win = Converter()
	win.mainloop()

if __name__ == "__main__":
	main()
```
In this case I have no loguru output for ZeroDivisionError (after button click) , only standard traceback. If I uncomment the last line in `__init__` (`#self.conversion_start()`) - I get what I need (loguru output for this exception when application is initialize).

How can I catch exceptions in the 'command'-function of tkinter-objects?
Sorry for my English.

**Comment by Delgan** [comment]
Hi @ppmm9999.

This is an interesting case. It seems exceptions are handled somehow internally by `tkinter`. They are not propagated to the main thread and hence can't be caught by `loguru` this way (this is also why the program do not crash when an error occurs).

It seems that the problem has been raised several times:
* [Why does an generic “try / except” block not apply to a Python TkInter callback?](https://stackoverflow.com/questions/37493309/why-does-an-generic-try-except-block-not-apply-to-a-python-tkinter-callback/37493588)
* [How can I make silent exceptions louder in tkinter?](https://stackoverflow.com/questions/4770993/how-can-i-make-silent-exceptions-louder-in-tkinter)
* [Handling exception in python tkinter](https://stackoverflow.com/questions/15246523/handling-exception-in-python-tkinter)
* [tkinter python: catching exceptions](https://stackoverflow.com/questions/6666882/tkinter-python-catching-exceptions/26363503)
* [Saving exceptions in Tkinter using Python](https://stackoverflow.com/questions/38464351/saving-exceptions-in-tkinter-using-python)


And so on. :p

Basically, you need to override the `report_callback_exception()`. You can use the [`logger.opt()`](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.opt) to log the error as if it has been caught by the `@catch` decorator. For example:

```python
class Converter(Tk):
	def __init__(self):
		Tk.__init__(self)
		self.formats=''
		self.button_start=Button(self, text="Start", command=self.conversion_start)
		self.button_start.pack()
		#self.conversion_start()

	def conversion_start(self):
		self.formats=RLCExcelFile()

	def report_callback_exception(self, *args):
		logger.opt(exception=args).error("Error in tkinter")
```

**Comment by ppmm9999** [comment]
Thanks, it works as I want
